### PR TITLE
add rstudio bash script

### PR DESCRIPTION
**Pull request by dgrassellyb**

### CODE REVIEW

File: .devcontainer/postCreateCommand.sh [hunk 1]
```diff
@@ -1,5 +1,10 @@
...
 
+# move script rstudio into /usr/bin
+cp "$(pwd)/.devcontainer/rstudio.sh" /usr/bin/rstudio
+
+# Restore renv and install staged dependencies
 R -q -e 'renv::activate(profile = paste(R.version$major, substr(R.version$minor, 1, 1), sep = ".")); renv::restore(); staged.dependencies::install_deps(staged.dependencies::dependency_table(project = ".", verbose = 1), verbose = 1);'
 
-jq --arg folder "$(pwd)/" '. + { "initial_working_directory": $folder }' .devcontainer/rstudio-prefs.json > ~/.config/rstudio/rstudio-prefs.json
+# Define rstudio default working directory
+jq --arg folder "$(pwd)/" '. + { "initial_working_directory": $folder }' .devcontainer/rstudio-prefs.json > ~/.config/rstudio/rstudio-prefs.json
```
Comment (galachad) [inline .devcontainer/postCreateCommand.sh:10]:
```suggestion
jq --arg folder "$(pwd)/" '. + { "initial_working_directory": $folder }' .devcontainer/rstudio-prefs.json > ~/.config/rstudio/rstudio-prefs.json

```

Add EOF.

File: .devcontainer/rstudio.sh [hunk 2]
```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+
+echo "Waiting for rstudio-server to be ready.."
+
+# Get rstudio url and open it
+url=$(jq -r ".CODESPACE_NAME" /workspaces/.codespaces/shared/environment-variables.json)
+url_rstudio="https://$url-8787.app.github.dev"
```
Comment (galachad) [inline .devcontainer/rstudio.sh:7]: The url is not possible to reach without gitlab auth.

I will propose to keep it simple.

`$BROWSER "https://${CODESPACE_NAME}-8787.${GITHUB_CODESPACES_PORT_FORWARDING_DOMAIN}"`

If you would like to check if rstudio is stareted we should query `http://localhost:8787`.

File: .devcontainer/postCreateCommand.sh [hunk 3]
```diff
@@ -1,5 +1,10 @@
 #!/bin/bash
 
+# move script rstudio into /usr/bin
+cp "$(pwd)/.devcontainer/rstudio.sh" /usr/bin/rstudio
```
Comment (galachad) [inline .devcontainer/postCreateCommand.sh:4]: We should also add `sudo` to cp and `chmod +x` to `/usr/bin/rstudio`

```suggestion
sudo cp "$(pwd)/.devcontainer/rstudio.sh" /usr/bin/rstudio
sudo chmod +x /usr/bin/rstudio
```

**Review by galachad — CHANGES_REQUESTED** [review]
I think we should keep the script as simple as possible.

File: .devcontainer/rstudio.sh [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    echo "Rstudio server ready ! You can now go to this url : $url_rstudio"
+    exit 0
+  else
+    sleep $WAIT_TIME
+    ((retries++))
+  fi
+done
+
+echo "Error accessing rstudio ($url_rstudio) - Timeout reached."
+exit 1
```
Comment (galachad) [inline .devcontainer/rstudio.sh:29]:
```suggestion
#!/bin/bash

$BROWSER "https://${CODESPACE_NAME}-8787.${GITHUB_CODESPACES_PORT_FORWARDING_DOMAIN}"

exit 1

```

**Comment by galachad** [comment]
Hi @dgrassellyb could you take a look on comments?

**Comment by dgrassellyb** [comment]
> Hi @dgrassellyb could you take a look on comments?

Hi @galachad this one is quite outdated, it's a bit duplicate on what you did for codespaces 
for example you're already doing this command in postAttachCommand `"postAttachCommand": "$BROWSER \"https://${CODESPACE_NAME}-8787.${GITHUB_CODESPACES_PORT_FORWARDING_DOMAIN}/\"",`

I see you suggested to delete the part waiting rstudio server to be ready (rstudio.sh script) - so I guess we can just close this PR

**Review by galachad — APPROVED** [review]
It will be nice to have the `rstudio` command in the console.

I'm merging this PR.

**Pull request merged**